networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -307,3 +308,41 @@ def _asr_do_remove_floating_ip(self, ri, floating_ip,
         self._edit_running_config(confstr,
                                   'REMOVE_STATIC_SRC_TRL_NO_VRF_MATCH')
         self._asr_do_remove_secondary_ip(ri, floating_ip, ex_gw_port)
+
+    def _do_set_snat_pool(self, pool_name, pool_start,
+                          pool_end, pool_net, is_delete):
+        try:
+            if is_delete:
+                conf_str = asr1k_snippets.DELETE_NAT_POOL % (
+                    pool_name, pool_start, pool_end, pool_net.netmask)
+                # TODO(update so that hosting device name is passed down)
+                self._edit_running_config(conf_str, 'DELETE_NAT_POOL')
+
+            else:
+                conf_str = asr1k_snippets.CREATE_NAT_POOL % (
+                    pool_name, pool_start, pool_end, pool_net.netmask)
+                # TODO(update so that hosting device name is passed down)
+                self._edit_running_config(conf_str, 'CREATE_NAT_POOL')
+        except Exception as cse:
+            LOG.error(_LE("Temporary disable NAT_POOL exception handling: "
+                          "%s"), cse)
+
+    def _set_snat_pools_from_hosting_info(self, ri, gw_port, is_delete):
+        # TODO(tbachma ): unique naming for more than one pool
+        vrf_name = self._get_vrf_name(ri)
+        for net_id in gw_port['hosting_info'].get('snat_subnet_ids', []):
+            for subnet in gw_port['extra_subnets']:
+                if subnet['id'] == net_id:
+                    net = netaddr.IPNetwork(subnet['cidr'])
+                    pool_start = str(netaddr.IPAddress(net.first + 2))
+                    pool_end = str(netaddr.IPAddress(net.last - 1))
+                    pool_name = "%s_nat_pool" % (vrf_name)
```

Comment:
the codes seem to suggest that there could be multiple SNAT pools however here the pool_name will always be the same...So instead of doing this half-baked multiple SNAT pools, maybe we just make a single SNAT pool solution solid then?